### PR TITLE
Update Bidirectional plugin to use expressionPlugin

### PR DESCRIPTION
**Pull request by jason-fox**
Related: https://github.com/telefonicaid/iotagent-node-lib/pull/1266#issuecomment-1217786005

This would remove the hardcoded `legacyParser` from the Bidirectionality plugin.

### CODE REVIEW

File: lib/plugins/bidirectionalData.js [hunk 1]
```diff
@@ -26,7 +26,7 @@
 const async = require('async');
```
Comment (fgalan) [inline lib/plugins/bidirectionalData.js:26]: An entry in CHANGES_NEXT_RELEASE file about the changes in this PR should be added.

File: lib/plugins/bidirectionalData.js [hunk 2]
```diff
@@ -181,16 +181,16 @@ function getReverseTransformations(list, values, callback) {
...
- * Apply a list of transformations to the reported values, 
+ * Apply a list of transformations to the reported values,
  * generating a new array of values with the additional data.
  * For NGSI-v2 this consists of value transformations only
  *
  * @param {Array} values                List of reported attributes (with their name, type and value).
  * @param {Array} transformations       List of transformations to apply (with their name, type and expression).
  */
-function processTransformationsNGSIv2(values, transformations, callback) {
+function processTransformationsNGSIv2(values, transformations, typeInformation, callback) {
```
Comment (fgalan) [inline lib/plugins/bidirectionalData.js:191]: `typeInformation` should be added as new `@param` in the function "JavaDoc"

File: lib/plugins/bidirectionalData.js [hunk 3]
```diff
@@ -208,22 +208,22 @@ function processTransformationsNGSIv2(values, transformations, callback) {
...
- * Apply a list of transformations to the reported values, 
+ * Apply a list of transformations to the reported values,
  * generating a new array of values with the additional data.
  * For NGSI-LD the output includes value, metadata and datasetId
  *
  * @param {Array} values                List of reported attributes (with their name, type and value).
  * @param {Array} transformations       List of transformations to apply (with their name, type and expression).
  */
-function processTransformationsNGSILD(values, transformations, callback) {
+function processTransformationsNGSILD(values, transformations, typeInformation, callback) {
```
Comment (fgalan) [inline lib/plugins/bidirectionalData.js:226]: `typeInformation` should be added as new `@param` in the function "JavaDoc"

File: lib/plugins/bidirectionalData.js [hunk 4]
```diff
@@ -289,11 +299,11 @@ function processTransformationsNGSILD(values, transformations, callback) {
  * @param {Array} values                List of reported attributes (with their name, type and value).
  * @param {Array} transformations       List of transformations to apply (with their name, type and expression).
  */
-function processTransformations(values, transformations, callback) {
-    if (config.checkNgsiLD({})){
-        processTransformationsNGSILD(values, transformations, callback)
+function processTransformations(values, typeInformation, transformations, callback) {
```
Comment (fgalan) [inline lib/plugins/bidirectionalData.js:302]: `typeInformation` should be added as new `@param` in the function "JavaDoc"

 Note in this case the secuence is values-typeInformation-transformations-callback but in the other two functions is values-transformations-typeInformation-callback. Is this intentional?

File: lib/plugins/bidirectionalData.js (hunk 1 above)
Comment (jason-fox) [reply lib/plugins/bidirectionalData.js:26]: Fixed e0220ae

File: lib/plugins/bidirectionalData.js (hunk 2 above)
Comment (jason-fox) [reply lib/plugins/bidirectionalData.js:192]: Fixed e0220ae

File: lib/plugins/bidirectionalData.js (hunk 3 above)
Comment (jason-fox) [reply lib/plugins/bidirectionalData.js:228]: Fixed e0220ae

File: lib/plugins/bidirectionalData.js (hunk 4 above)
Comment (jason-fox) [reply lib/plugins/bidirectionalData.js]: Fixed e0220ae and aligned the params for consistency and also aligned the calling code at line 360

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Comment by jason-fox** [comment]
> LGTM. Thanks!

Except that broke the tests, so I guess the ordering for the first call needs to be kept. Fixed 98197de

**Pull request merged**